### PR TITLE
[tricycle_controller] Use generate_parameter_library

### PR DESCRIPTION
**Pull request by christophfroehlich**
As per title.

- use generate_parameter_library, remove not anymore needed checks and variables, rewrite tests
- let the load_controller test use the yaml from the test file

I also updated the documentation a bit, see #850 

This breaks ABI because of the class variable change, is this a bad idea to backport?

### CODE REVIEW

File: tricycle_controller/src/tricycle_controller.cpp [hunk 1]
```diff
@@ -357,7 +291,7 @@ CallbackReturn TricycleController::on_configure(const rclcpp_lifecycle::State &
   }
 
   // initialize command subscriber
-  if (use_stamped_vel_)
+  if (params_.use_stamped_vel)
```
Comment (bmagyar) [inline tricycle_controller/src/tricycle_controller.cpp:294]: probably due to be removed soon too

File: tricycle_controller/src/tricycle_controller_parameter.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
+tricycle_controller:
+  traction_joint_name: {
+    type: string,
+    default_value: "",
+    read_only: true,
```
Comment (bmagyar) [inline tricycle_controller/src/tricycle_controller_parameter.yaml:5]: do we want read_only-s here?

Comment (christophfroehlich) [reply tricycle_controller/src/tricycle_controller_parameter.yaml:5]: after reporting #966, no..

File: tricycle_controller/src/tricycle_controller.cpp (hunk 1 above)
Comment (christophfroehlich) [reply tricycle_controller/src/tricycle_controller.cpp:294]: in this PR or in a follow up?

**Review by bmagyar — APPROVED** [review]

**Pull request merged**